deploy: update CSI sidecars to latest versions available

iPraveenParihar opened the pull request:
<!-- Please take a look at our [Contributing](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#Code-contribution-workflow)
documentation before submitting a Pull Request!
Thank you for contributing to ceph-csi! -->

# Describe what this PR does #

1. Below sidecars are updated with latest available versions
```
csi-attacher: v4.10.0
csi-snapshotter: v8.4.0
csi-resizer: v2.0.0
csi-provisioner: v6.0.0
csi-node-driver-registrar: v2.15.0
```

2. deploy: include '--automaxprocs' to csi sidecars

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

iPraveenParihar committed:
deploy: update CSI sidecars to latest versions available

Below sidecars are updated with latest available versions

csi-attacher: v4.10.0
csi-snapshotter: v8.4.0
csi-resizer: v2.0.0
csi-provisioner: v6.0.0
csi-node-driver-registrar: v2.15.0

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
deploy: include '--automaxprocs' to csi sidecars

Signed-off-by: Praveen M <m.praveen@ibm.com>

Rakshith-R commented on charts/ceph-csi-rbd/README.md:
```diff
@@ -111,7 +111,7 @@ charts and their default values.
 | `nodeplugin.profiling.enabled`                 | Specifies whether profiling should be enabled                                                                                                        | `false`                                            |
 | `nodeplugin.registrar.image.repository`        | Node Registrar image repository URL                                                                                                                  | `registry.k8s.io/sig-storage/csi-node-driver-registrar` |
-| `nodeplugin.registrar.image.tag`               | Image tag                                                                                                                                            | `v2.13.0`                                           |
+| `nodeplugin.registrar.image.tag`               | Image tag                                                                                                                                            | `v2.14.0`                                           |
```
attacher image needs to be updated in this file

Rakshith-R reviewed (commented):
Please update image versions in ceph-csi-op repo too.

iPraveenParihar commented:
> Please update image versions in ceph-csi-op repo too.

They were already updated here https://github.com/ceph/ceph-csi-operator/pull/252.
Will have to update in Rook :)

nixpanic commented:
> They were already updated here [ceph/ceph-csi-operator#252](https://github.com/ceph/ceph-csi-operator/pull/252). Will have to update in Rook :)

https://github.com/ceph/ceph-csi-operator/blob/90402565977568471446eb4992b08d033b6ce9eb/internal/controller/defaults.go#L34

csi-snapshotter still needs to be updated there.

Madhu-1 commented:
@mergifyio queue

Madhu-1 commented:
@iPraveenParihar can you please check , looks like some real issue as all test failed.

iPraveenParihar commented:
Helm deployments are failing because of csi-provisioner.


I have sent a fix PR - https://github.com/kubernetes-csi/external-provisioner/pull/1397

Rakshith-R commented:
> Helm deployments are failing because of csi-provisioner.
> 
> I have sent a fix PR - [kubernetes-csi/external-provisioner#1397](https://github.com/kubernetes-csi/external-provisioner/pull/1397)

@iPraveenParihar , Let's comment out the flag with a TODO and open a tracker issue for the same.
then we can merge this pr ?

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.31

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

iPraveenParihar commented:
/test ci/centos/k8s-e2e-external-storage/1.31

iPraveenParihar commented:
> > Helm deployments are failing because of csi-provisioner.
> > I have sent a fix PR - [kubernetes-csi/external-provisioner#1397](https://github.com/kubernetes-csi/external-provisioner/pull/1397)
> 
> @iPraveenParihar , Let's comment out the flag with a TODO and open a tracker issue for the same. then we can merge this pr ?

done, updated csi-provisioner to previous version v5.2.0.
Lets see if tests passes

iPraveenParihar commented:
both failed because of CephFS VolumeGroupSnapshot

https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/job/mini-e2e-helm_k8s-1.31/551/display/redirect

```
  I0716 13:12:10.624922 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (0 seconds elapsed)
  I0716 13:12:12.625423 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (2 seconds elapsed)
  I0716 13:12:12.628300 81584 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7416-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-9343982a-da19-4c58-8b6f-6aa6c6045045 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-8a55b5dd-1df8-4bc4-8c51-f2f7b13e5ecd 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-c5de278e-1096-4482-a864-7617a527df58]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:12:14.626467 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (4 seconds elapsed)
  I0716 13:12:14.629711 81584 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7416-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-9343982a-da19-4c58-8b6f-6aa6c6045045 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-8a55b5dd-1df8-4bc4-8c51-f2f7b13e5ecd 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-c5de278e-1096-4482-a864-7617a527df58]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:12:16.626711 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (6 seconds elapsed)
  I0716 13:12:16.629820 81584 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7416-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-9343982a-da19-4c58-8b6f-6aa6c6045045 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-8a55b5dd-1df8-4bc4-8c51-f2f7b13e5ecd 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-c5de278e-1096-4482-a864-7617a527df58]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:12:18.625723 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (8 seconds elapsed)
  I0716 13:12:18.628719 81584 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7416-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-9343982a-da19-4c58-8b6f-6aa6c6045045 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-8a55b5dd-1df8-4bc4-8c51-f2f7b13e5ecd 0001-0024-59986703-7528-4649-9ede-634caa229656-0000000000000001-c5de278e-1096-4482-a864-7617a527df58]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:12:20.625652 81584 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7416-vgs (10 seconds elapsed)
  ```
  ```
I0716 13:02:22.029517 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (0 seconds elapsed)
  I0716 13:02:24.030295 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (2 seconds elapsed)
  I0716 13:02:24.032779 84328 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7278-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-5ebf21ce-a4e7-4023-adb4-ddf842b3df21 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-28edc411-e556-498f-ac55-398cbc011ada 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-b226f913-15b4-402f-afa7-74680bc85126]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:02:26.030947 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (4 seconds elapsed)
  I0716 13:02:26.033576 84328 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7278-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-5ebf21ce-a4e7-4023-adb4-ddf842b3df21 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-28edc411-e556-498f-ac55-398cbc011ada 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-b226f913-15b4-402f-afa7-74680bc85126]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:02:28.030257 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (6 seconds elapsed)
  I0716 13:02:28.032752 84328 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7278-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-5ebf21ce-a4e7-4023-adb4-ddf842b3df21 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-28edc411-e556-498f-ac55-398cbc011ada 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-b226f913-15b4-402f-afa7-74680bc85126]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:02:30.029635 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (8 seconds elapsed)
  I0716 13:02:30.032070 84328 volumegroupsnapshot_base.go:372] current state of VolumeGroupSnapshot cephfs-7278-vgs. ReadyToUse: false, Error: Failed to check and update group snapshot content: failed to take group snapshot of the volumes [0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-5ebf21ce-a4e7-4023-adb4-ddf842b3df21 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-28edc411-e556-498f-ac55-398cbc011ada 0001-0024-86362b95-b906-4dc6-bf0d-1fd8f1e57077-0000000000000001-b226f913-15b4-402f-afa7-74680bc85126]: "rpc error: code = Internal desc = quiesce operation is in progress"
  I0716 13:02:32.030278 84328 volumegroupsnapshot_base.go:344] waiting for VolumeGroupSnapshot cephfs-7278-vgs (10 seconds elapsed)
  ```
  
🤔 will check what the issue is

iPraveenParihar commented:
did a quick check with my machine with snapshotter: v8.3.0 and seems to be fine.

```
[pm@dhcp53-115 cephfs]$ k get pvc
NAME           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
cephfs-pvc-1   Bound    pvc-fdcee156-c83f-4df1-80e3-689776440e88   1Gi        RWO            rook-cephfs    <unset>                 22m
cephfs-pvc-2   Bound    pvc-d7f81f2b-59d8-40d6-9914-4d7bd4814c9a   1Gi        RWO            rook-cephfs    <unset>                 22m
cephfs-pvc-3   Bound    pvc-bdea7211-b3d2-4547-8237-6dc96fd3274f   1Gi        RWO            rook-cephfs    <unset>                 22m

[pm@dhcp53-115 cephfs]$ k get po
NAME    READY   STATUS    RESTARTS   AGE
pod-1   1/1     Running   0          19m
pod-2   1/1     Running   0          19m
pod-3   1/1     Running   0          19m

[pm@dhcp53-115 cephfs]$ k get vgs
NAME                   READYTOUSE   VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTCONTENT                              CREATIONTIME   AGE
cephfs-groupsnapshot   true         csi-cephfsplugin-groupsnapclass   groupsnapcontent-3589aff7-fd17-46bf-9647-189c19ad8ba4   6m52s          6m54s
[pm@dhcp53-115 cephfs]$ k get vgsc
NAME                                                    READYTOUSE   DELETIONPOLICY   DRIVER                          VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTNAMESPACE   VOLUMEGROUPSNAPSHOT    AGE
groupsnapcontent-3589aff7-fd17-46bf-9647-189c19ad8ba4   true         Delete           rook-ceph.cephfs.csi.ceph.com   csi-cephfsplugin-groupsnapclass   default                        cephfs-groupsnapshot   6m57s

[pm@dhcp53-115 cephfs]$ k get vs
NAME                                                                        READYTOUSE   SOURCEPVC      SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS   SNAPSHOTCONTENT                                                                CREATIONTIME   AGE
snapshot-0752220023c83046be2d541a5ee47c814587371a533c3c12c70d9eaba81bb830   true         cephfs-pvc-1                                                         snapcontent-0752220023c83046be2d541a5ee47c814587371a533c3c12c70d9eaba81bb830   6m53s          6m58s
snapshot-9ca10acd2c82377db4d457b2584958e31b9fcc4806fe589ee6100aef05818165   true         cephfs-pvc-2                                                         snapcontent-9ca10acd2c82377db4d457b2584958e31b9fcc4806fe589ee6100aef05818165   6m49s          6m53s
snapshot-a80841d7a342c067345e8899c57740112583104e5e2389f15d0fe3502717fbbd   true         cephfs-pvc-3                                                         snapcontent-a80841d7a342c067345e8899c57740112583104e5e2389f15d0fe3502717fbbd   6m51s          6m56s

[pm@dhcp53-115 cephfs]$ k get volumesnapshotcontents.snapshot.storage.k8s.io
NAME                                                                           READYTOUSE   RESTORESIZE   DELETIONPOLICY   DRIVER                          VOLUMESNAPSHOTCLASS   VOLUMESNAPSHOT                                                              VOLUMESNAPSHOTNAMESPACE   AGE
snapcontent-0752220023c83046be2d541a5ee47c814587371a533c3c12c70d9eaba81bb830   true                       Delete           rook-ceph.cephfs.csi.ceph.com                         snapshot-0752220023c83046be2d541a5ee47c814587371a533c3c12c70d9eaba81bb830   default                   7m4s
snapcontent-9ca10acd2c82377db4d457b2584958e31b9fcc4806fe589ee6100aef05818165   true                       Delete           rook-ceph.cephfs.csi.ceph.com                         snapshot-9ca10acd2c82377db4d457b2584958e31b9fcc4806fe589ee6100aef05818165   default                   7m
snapcontent-a80841d7a342c067345e8899c57740112583104e5e2389f15d0fe3502717fbbd   true                       Delete           rook-ceph.cephfs.csi.ceph.com                         snapshot-a80841d7a342c067345e8899c57740112583104e5e2389f15d0fe3502717fbbd   default                   7m2s
```

iPraveenParihar commented:
failed job `Internal desc = quiesce operation is in progress` seems to have struck at quiesce.
Will take a look at ceph logs, if available and update.

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar commented:
seems to be some problem with snapshotter
```
  [38;5;9m[FAILED] failed to test volumeGroupSnapshot: failed to create clones: failed to get VolumeSnapshot: volumesnapshots.snapshot.storage.k8s.io "snapshot-a400cafa8cbdd8eb2c59da70a98fc4d46529598d69a12e3c668cab24958a4886" not found[0m
  [38;5;9mIn [1m[It][0m[38;5;9m at: [1m/go/src/github.com/ceph/ceph-csi/e2e/utils.go:2014[0m [38;5;243m@ 08/26/25 16:16:45.823[0m
```

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar commented:
Looks like there is some issue with 8.3.0 external-snapshotter

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar committed:
deploy: update groupsnapshot.storage.k8s.io/v1beta1 to v1beta2

With external-snapshotter v8.4.0 the v1beta1 VolumeGroupSnapshot API
is now marked as deprecated. and v1beta2 is introduced.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar commented:
[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.33/runs/310/nodes/90/log/?start=0)

```
  I1028 18:24:18.378748   76504 warnings.go:110] "Warning: groupsnapshot.storage.k8s.io/v1beta1 VolumeGroupSnapshot is deprecated; use groupsnapshot.storage.k8s.io/v1beta2 VolumeGroupSnapshot"
  I1028 18:24:18.382117   76504 warnings.go:110] "Warning: groupsnapshot.storage.k8s.io/v1beta1 VolumeGroupSnapshotContent is deprecated; use groupsnapshot.storage.k8s.io/v1beta2 VolumeGroupSnapshotContent"
  I1028 18:24:18.382189 76504 utils.go:2031] failed to test volumeGroupSnapshot: failed to create clones: failed to get VolumeGroupSnapshotContent: conversion webhook for groupsnapshot.storage.k8s.io/v1beta2, Kind=VolumeGroupSnapshotContent failed: Post "https://snapshot-conversion-webhook-service.default.svc:443/convert?timeout=30s": service "snapshot-conversion-webhook-service" not found
```

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33

iPraveenParihar committed:
e2e: update external-snapshotter client to v8.4.0

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33/cephfs

black-dragon74 commented:
> [logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.33/runs/310/nodes/90/log/?start=0)
> 
> ```
>   I1028 18:24:18.378748   76504 warnings.go:110] "Warning: groupsnapshot.storage.k8s.io/v1beta1 VolumeGroupSnapshot is deprecated; use groupsnapshot.storage.k8s.io/v1beta2 VolumeGroupSnapshot"
>   I1028 18:24:18.382117   76504 warnings.go:110] "Warning: groupsnapshot.storage.k8s.io/v1beta1 VolumeGroupSnapshotContent is deprecated; use groupsnapshot.storage.k8s.io/v1beta2 VolumeGroupSnapshotContent"
>   I1028 18:24:18.382189 76504 utils.go:2031] failed to test volumeGroupSnapshot: failed to create clones: failed to get VolumeGroupSnapshotContent: conversion webhook for groupsnapshot.storage.k8s.io/v1beta2, Kind=VolumeGroupSnapshotContent failed: Post "https://snapshot-conversion-webhook-service.default.svc:443/convert?timeout=30s": service "snapshot-conversion-webhook-service" not found
> ```

It is expecting the conversion webhook to be deployed

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.33/cephfs

iPraveenParihar commented:
> It is expecting the conversion webhook to be deployed

That was before using 8.2.0 client, now I have updated client to 8.4.0
should pass now.

iPraveenParihar commented:
passed Cephfs [logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/job/mini-e2e_k8s-1.33-cephfs/3/display/redirect)

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.34

Madhu-1 commented on charts/ceph-csi-cephfs/README.md:
```diff
@@ -138,7 +138,7 @@ charts and their default values.
 | `nodeplugin.profiling.enabled`                 | Specifies whether profiling should be enabled                                                                                                        | `false`                                            |
 | `nodeplugin.registrar.image.repository`        | Node-Registrar image repository URL                                                                                                                  | `registry.k8s.io/sig-storage/csi-node-driver-registrar` |
-| `nodeplugin.registrar.image.tag`               | Image tag                                                                                                                                            | `v2.13.0`                                           |
+| `nodeplugin.registrar.image.tag`               | Image tag                                                                                                                                            | `v2.15.0`                                           |
```
couple of sidecar update are missing in the doc

Madhu-1 commented on charts/ceph-csi-cephfs/values.yaml:
```diff
@@ -272,7 +272,7 @@ provisioner:
     image:
       repository: registry.k8s.io/sig-storage/csi-resizer
-      tag: v1.13.1
+      tag: v2.0.0
```
attacher update is missing here

Madhu-1 commented on deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:
```diff
@@ -93,7 +93,7 @@ spec:
               mountPath: /etc/ceph-csi-encryption-kms-config/
         - name: csi-provisioner
-          image: registry.k8s.io/sig-storage/csi-provisioner:v5.1.0
+          image: registry.k8s.io/sig-storage/csi-provisioner:v6.0.0
```
attacher update is missing here

Madhu-1 reviewed (changes requested):
small nits

iPraveenParihar commented:
will rebase after https://github.com/ceph/ceph-csi/pull/5718 is merged

black-dragon74 reviewed (dismissed):
Thank you! Seems using `v1beta2` saved us from deploying an extra conversion web hook 🎉

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio queue

iPraveenParihar commented:
@Mergifyio queue

iPraveenParihar commented:
@Mergifyio requeue

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.32

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
/retest ci/centos/mini-e2e/k8s-1.33

iPraveenParihar commented:
/retest ci/centos/upgrade-tests-cephfs

iPraveenParihar commented:
/retest ci/centos/upgrade-tests-rbd

Madhu-1 commented:
This is going crazy.

@mergifyqio dequeue

Madhu-1 commented:
@mergifyio dequeue

Madhu-1 commented:
@iPraveenParihar can you please check why CI is failing on this one

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
> @iPraveenParihar can you please check why CI is failing on this one

```
  I1107 07:55:29.136344 59122 volume_resource.go:323] Unexpected error: 
      <*errors.errorString | 0xc000b6e390>: 
      PersistentVolumeClaims [cephfs.csi.ceph.comv47r2] not all in phase Bound within 5m0s
      {
          s: "PersistentVolumeClaims [cephfs.csi.ceph.comv47r2] not all in phase Bound within 5m0s",
      }

############ PVC created after test failed ❌

I1107 07:55:34.308818       1 utils.go:347] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 GRPC call: /csi.v1.Controller/CreateVolume
I1107 07:55:34.308886       1 utils.go:348] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 GRPC request: {"capacity_range":{"required_bytes":1073741824},"name":"pvc-1096f753-d5d4-406e-af41-00ab42b9f303","parameters":{"clusterID":"8298d517-2d03-437a-8ace-8efc9d0e8440","csi.storage.k8s.io/pv/name":"pvc-1096f753-d5d4-406e-af41-00ab42b9f303","csi.storage.k8s.io/pvc/name":"cephfs.csi.ceph.comv47r2","csi.storage.k8s.io/pvc/namespace":"volumemode-7786","fsName":"myfs"},"secrets":"***stripped***","volume_capabilities":[{"access_mode":{"mode":"SINGLE_NODE_MULTI_WRITER"},"mount":{}}]}
W1107 07:55:34.308911       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
I1107 07:55:34.311357       1 omap.go:89] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 got omap values: (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): map[]
W1107 07:55:34.311379       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
I1107 07:55:34.319097       1 omap.go:159] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): map[csi.volume.pvc-1096f753-d5d4-406e-af41-00ab42b9f303:718d05df-d76f-4a84-9571-6174fa0a6e01])
I1107 07:55:34.321569       1 omap.go:159] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volume.718d05df-d76f-4a84-9571-6174fa0a6e01"): map[csi.imagename:csi-vol-718d05df-d76f-4a84-9571-6174fa0a6e01 csi.volname:pvc-1096f753-d5d4-406e-af41-00ab42b9f303 csi.volume.owner:volumemode-7786])
I1107 07:55:34.321592       1 fsjournal.go:319] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 Generated Volume ID (0001-0024-8298d517-2d03-437a-8ace-8efc9d0e8440-0000000000000001-718d05df-d76f-4a84-9571-6174fa0a6e01) and subvolume name (csi-vol-718d05df-d76f-4a84-9571-6174fa0a6e01) for request name (pvc-1096f753-d5d4-406e-af41-00ab42b9f303)
I1107 07:55:34.419662       1 controllerserver.go:482] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 cephfs: successfully created backing volume named csi-vol-718d05df-d76f-4a84-9571-6174fa0a6e01 for request name pvc-1096f753-d5d4-406e-af41-00ab42b9f303
I1107 07:55:34.420229       1 utils.go:354] ID: 27 Req-ID: pvc-1096f753-d5d4-406e-af41-00ab42b9f303 GRPC response: {"volume":{"capacity_bytes":1073741824,"volume_context":{"clusterID":"8298d517-2d03-437a-8ace-8efc9d0e8440","fsName":"myfs","subvolumeName":"csi-vol-718d05df-d76f-4a84-9571-6174fa0a6e01","subvolumePath":"/volumes/csi/csi-vol-718d05df-d76f-4a84-9571-6174fa0a6e01/6e723fd3-227b-4a07-aa60-9a9bd9bbbbd2"},"volume_id":"0001-0024-8298d517-2d03-437a-8ace-8efc9d0e8440-0000000000000001-718d05df-d76f-4a84-9571-6174fa0a6e01"}}

```

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
again its the same 😢 
```
  I1110 15:09:02.445710 60615 pv.go:806] PersistentVolumeClaim cephfs.csi.ceph.com6wrjm found but phase is Pending instead of Bound.
  I1110 15:09:04.446738 60615 volume_resource.go:323] Unexpected error: 
      <*errors.errorString | 0xc0011759b0>: 
      PersistentVolumeClaims [cephfs.csi.ceph.com6wrjm] not all in phase Bound within 5m0s
      {
          s: "PersistentVolumeClaims [cephfs.csi.ceph.com6wrjm] not all in phase Bound within 5m0s",
      }
  [38;5;9m[FAILED] PersistentVolumeClaims [cephfs.csi.ceph.com6wrjm] not all in phase Bound within 5m0s[0m
```

csi-cephfsplugin
```
I1110 15:04:03.977264       1 utils.go:347] ID: 19 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC call: /csi.v1.Controller/CreateVolume
I1110 15:04:03.977347       1 utils.go:348] ID: 19 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC request: {"capacity_range":{"required_bytes":1073741824},"name":"pvc-6c351d36-91ff-45f3-980b-a5d1a907b723","parameters":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","csi.storage.k8s.io/pv/name":"pvc-6c351d36-91ff-45f3-980b-a5d1a907b723","csi.storage.k8s.io/pvc/name":"cephfs.csi.ceph.com6wrjm","csi.storage.k8s.io/pvc/namespace":"volume-99","fsName":"myfs"},"secrets":"***stripped***","volume_capabilities":[{"access_mode":{"mode":"SINGLE_NODE_MULTI_WRITER"},"mount":{}}]}
W1110 15:04:03.977392       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
E1110 15:04:03.977632       1 controllerserver.go:326] ID: 19 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 validation and extraction of volume options failed: failed to fetch monitor list using clusterID (75c3a4d5-2a91-47c4-a76b-716eff9da391): missing configuration for cluster ID: "75c3a4d5-2a91-47c4-a76b-716eff9da391"
E1110 15:04:03.977669       1 utils.go:352] ID: 19 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC error: rpc error: code = InvalidArgument desc = failed to fetch monitor list using clusterID (75c3a4d5-2a91-47c4-a76b-716eff9da391): missing configuration for cluster ID: "75c3a4d5-2a91-47c4-a76b-716eff9da391"
I1110 15:04:34.764893       1 utils.go:347] ID: 20 GRPC call: /csi.v1.Identity/Probe
I1110 15:04:34.764919       1 utils.go:348] ID: 20 GRPC request: {}
I1110 15:04:34.764936       1 utils.go:354] ID: 20 GRPC response: {}
I1110 15:05:34.724124       1 utils.go:347] ID: 21 GRPC call: /csi.v1.Identity/Probe
I1110 15:05:34.724153       1 utils.go:348] ID: 21 GRPC request: {}
I1110 15:05:34.724421       1 utils.go:354] ID: 21 GRPC response: {}
I1110 15:06:34.723329       1 utils.go:347] ID: 22 GRPC call: /csi.v1.Identity/Probe
I1110 15:06:34.723356       1 utils.go:348] ID: 22 GRPC request: {}
I1110 15:06:34.723373       1 utils.go:354] ID: 22 GRPC response: {}
I1110 15:07:34.733896       1 utils.go:347] ID: 23 GRPC call: /csi.v1.Identity/Probe
I1110 15:07:34.733920       1 utils.go:348] ID: 23 GRPC request: {}
I1110 15:07:34.733931       1 utils.go:354] ID: 23 GRPC response: {}
I1110 15:08:34.722918       1 utils.go:347] ID: 24 GRPC call: /csi.v1.Identity/Probe
I1110 15:08:34.722952       1 utils.go:348] ID: 24 GRPC request: {}
I1110 15:08:34.722964       1 utils.go:354] ID: 24 GRPC response: {}
I1110 15:09:04.556645       1 utils.go:347] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf GRPC call: /csi.v1.Controller/CreateVolume
I1110 15:09:04.556723       1 utils.go:348] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf GRPC request: {"capacity_range":{"required_bytes":1073741824},"name":"pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf","parameters":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","csi.storage.k8s.io/pv/name":"pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf","csi.storage.k8s.io/pvc/name":"cephfs.csi.ceph.comzxpjq","csi.storage.k8s.io/pvc/namespace":"volume-expand-7820","fsName":"myfs"},"secrets":"***stripped***","volume_capabilities":[{"access_mode":{"mode":"SINGLE_NODE_MULTI_WRITER"},"mount":{}}]}
W1110 15:09:04.556744       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
E1110 15:09:04.576085       1 omap.go:80] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf omap not found (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): rados: ret=-2, No such file or directory
W1110 15:09:04.576120       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
I1110 15:09:04.582845       1 omap.go:159] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): map[csi.volume.pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf:c700409f-58b2-4819-9387-ce7ef9851078])
I1110 15:09:04.585378       1 omap.go:159] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volume.c700409f-58b2-4819-9387-ce7ef9851078"): map[csi.imagename:csi-vol-c700409f-58b2-4819-9387-ce7ef9851078 csi.volname:pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf csi.volume.owner:volume-expand-7820])
I1110 15:09:04.585403       1 fsjournal.go:319] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf Generated Volume ID (0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078) and subvolume name (csi-vol-c700409f-58b2-4819-9387-ce7ef9851078) for request name (pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf)
I1110 15:09:04.700031       1 controllerserver.go:482] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf cephfs: successfully created backing volume named csi-vol-c700409f-58b2-4819-9387-ce7ef9851078 for request name pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf
I1110 15:09:04.700217       1 utils.go:354] ID: 25 Req-ID: pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf GRPC response: {"volume":{"capacity_bytes":1073741824,"volume_context":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","fsName":"myfs","subvolumeName":"csi-vol-c700409f-58b2-4819-9387-ce7ef9851078","subvolumePath":"/volumes/csi/csi-vol-c700409f-58b2-4819-9387-ce7ef9851078/115f4fe2-4c57-402b-b11e-5ea005a48089"},"volume_id":"0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078"}}
I1110 15:09:06.687199       1 utils.go:347] ID: 26 Req-ID: 0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078 GRPC call: /csi.v1.Controller/ControllerPublishVolume
I1110 15:09:06.687261       1 utils.go:348] ID: 26 Req-ID: 0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078 GRPC request: {"node_id":"minikube","volume_capability":{"access_mode":{"mode":"SINGLE_NODE_MULTI_WRITER"},"mount":{}},"volume_context":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","fsName":"myfs","storage.kubernetes.io/csiProvisionerIdentity":"1762787007174-4720-cephfs.csi.ceph.com","subvolumeName":"csi-vol-c700409f-58b2-4819-9387-ce7ef9851078","subvolumePath":"/volumes/csi/csi-vol-c700409f-58b2-4819-9387-ce7ef9851078/115f4fe2-4c57-402b-b11e-5ea005a48089"},"volume_id":"0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078"}
W1110 15:09:06.687361       1 controllerserver.go:1162] ID: 26 Req-ID: 0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078 controller publish secret not found: controller publish secret name or namespace is empty in csi config file for cluster 75c3a4d5-2a91-47c4-a76b-716eff9da391
I1110 15:09:06.687392       1 utils.go:354] ID: 26 Req-ID: 0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-c700409f-58b2-4819-9387-ce7ef9851078 GRPC response: {}
I1110 15:09:09.542981       1 utils.go:347] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC call: /csi.v1.Controller/CreateVolume
I1110 15:09:09.543336       1 utils.go:348] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC request: {"capacity_range":{"required_bytes":1073741824},"name":"pvc-6c351d36-91ff-45f3-980b-a5d1a907b723","parameters":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","csi.storage.k8s.io/pv/name":"pvc-6c351d36-91ff-45f3-980b-a5d1a907b723","csi.storage.k8s.io/pvc/name":"cephfs.csi.ceph.com6wrjm","csi.storage.k8s.io/pvc/namespace":"volume-99","fsName":"myfs"},"secrets":"***stripped***","volume_capabilities":[{"access_mode":{"mode":"SINGLE_NODE_MULTI_WRITER"},"mount":{}}]}
W1110 15:09:09.543366       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
I1110 15:09:09.545863       1 omap.go:89] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 got omap values: (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): map[]
W1110 15:09:09.545881       1 credentials.go:57] adminID and adminKey are deprecated, please use userID and userKey instead
I1110 15:09:09.551227       1 omap.go:159] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volumes.default"): map[csi.volume.pvc-6c351d36-91ff-45f3-980b-a5d1a907b723:1c6d504d-8537-4707-9060-f07395182a25])
I1110 15:09:09.553783       1 omap.go:159] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 set omap keys (pool="myfs-metadata", namespace="csi", name="csi.volume.1c6d504d-8537-4707-9060-f07395182a25"): map[csi.imagename:csi-vol-1c6d504d-8537-4707-9060-f07395182a25 csi.volname:pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 csi.volume.owner:volume-99])
I1110 15:09:09.553804       1 fsjournal.go:319] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 Generated Volume ID (0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-1c6d504d-8537-4707-9060-f07395182a25) and subvolume name (csi-vol-1c6d504d-8537-4707-9060-f07395182a25) for request name (pvc-6c351d36-91ff-45f3-980b-a5d1a907b723)
I1110 15:09:09.650641       1 controllerserver.go:482] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 cephfs: successfully created backing volume named csi-vol-1c6d504d-8537-4707-9060-f07395182a25 for request name pvc-6c351d36-91ff-45f3-980b-a5d1a907b723
I1110 15:09:09.650754       1 utils.go:354] ID: 27 Req-ID: pvc-6c351d36-91ff-45f3-980b-a5d1a907b723 GRPC response: {"volume":{"capacity_bytes":1073741824,"volume_context":{"clusterID":"75c3a4d5-2a91-47c4-a76b-716eff9da391","fsName":"myfs","subvolumeName":"csi-vol-1c6d504d-8537-4707-9060-f07395182a25","subvolumePath":"/volumes/csi/csi-vol-1c6d504d-8537-4707-9060-f07395182a25/c1c0fcf0-508d-47ff-abec-834d7a8dbb6c"},"volume_id":"0001-0024-75c3a4d5-2a91-47c4-a76b-716eff9da391-0000000000000001-1c6d504d-8537-4707-9060-f07395182a25"}}
```

csi-provisioner
```
I1110 15:04:03.975113       1 event.go:389] "Event occurred" object="volume-99/cephfs.csi.ceph.com6wrjm" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="Provisioning" message="External provisioner is provisioning volume for claim \"volume-99/cephfs.csi.ceph.com6wrjm\""
I1110 15:04:03.977910       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=0
E1110 15:04:03.977943       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": rpc error: code = InvalidArgument desc = failed to fetch monitor list using clusterID (75c3a4d5-2a91-47c4-a76b-716eff9da391): missing configuration for cluster ID: \"75c3a4d5-2a91-47c4-a76b-716eff9da391\"" logger="UnhandledError"
I1110 15:04:03.977981       1 event.go:389] "Event occurred" object="volume-99/cephfs.csi.ceph.com6wrjm" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Warning" reason="ProvisioningFailed" message="Volume provisioning failed with infeasible error. Retries will be delayed. rpc error: code = InvalidArgument desc = failed to fetch monitor list using clusterID (75c3a4d5-2a91-47c4-a76b-716eff9da391): missing configuration for cluster ID: \"75c3a4d5-2a91-47c4-a76b-716eff9da391\""
I1110 15:04:04.478679       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=1
E1110 15:04:04.478726       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:04:05.479515       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=2
E1110 15:04:05.479567       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:04:07.480263       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=3
E1110 15:04:07.480322       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:04:11.480728       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=4
E1110 15:04:11.480774       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:04:19.481237       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=5
E1110 15:04:19.481291       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:04:35.481752       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=6
E1110 15:04:35.481799       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:05:07.482678       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=7
E1110 15:05:07.482731       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:06:11.483377       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=8
E1110 15:06:11.483439       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:08:19.483839       1 controller.go:986] "Retrying syncing claim" key="6c351d36-91ff-45f3-980b-a5d1a907b723" failures=9
E1110 15:08:19.483904       1 controller.go:1009] "Unhandled Error" err="error syncing claim \"6c351d36-91ff-45f3-980b-a5d1a907b723\": skipping volume provisioning for pvc 6c351d36-91ff-45f3-980b-a5d1a907b723, because provisioning previously failed with infeasible error" logger="UnhandledError"
I1110 15:09:04.554948       1 event.go:389] "Event occurred" object="volume-expand-7820/cephfs.csi.ceph.comzxpjq" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="Provisioning" message="External provisioner is provisioning volume for claim \"volume-expand-7820/cephfs.csi.ceph.comzxpjq\""
I1110 15:09:04.705176       1 event.go:389] "Event occurred" object="volume-expand-7820/cephfs.csi.ceph.comzxpjq" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="ProvisioningSucceeded" message="Successfully provisioned volume pvc-6dbc24e9-e3e4-4c80-a01c-d5d1e5a476cf"
I1110 15:09:09.540999       1 event.go:389] "Event occurred" object="volume-99/cephfs.csi.ceph.com6wrjm" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="Provisioning" message="External provisioner is provisioning volume for claim \"volume-99/cephfs.csi.ceph.com6wrjm\""
```

trying with adding more timeout to account for error
```
Timeouts:
  ClaimProvision: 10m
```

iPraveenParihar commented:
/test ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
```
error: error execution phase wait-control-plane: failed while waiting for the control plane to start: kube-apiserver check failed at https://192.168.58.2:8443/livez: Get "https://control-plane.minikube.internal:8443/livez?timeout=10s": dial tcp 192.168.58.2:8443: connect: connection refused

To see the stack trace of this error execute with --v=5 or higher
```

seems to be jenkins connection issue 😕

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
/test ci/centos/k8s-e2e-external-storage/1.33

Madhu-1 commented:
@iPraveenParihar if the CI is not stable in this PR, merging this will cause issues, we need to make sure the tests are stable with this PR

iPraveenParihar commented:
> @iPraveenParihar if the CI is not stable in this PR, merging this will cause issues, we need to make sure the tests are stable with this PR

only k8s 1.34 is failing at deploy k8s
```
stderr:

	[WARNING SystemVerification]: kernel release 5.14.0-612.el9.x86_64 is unsupported. Supported LTS versions from the 5.x series are 5.4, 5.10 and 5.15. Any 6.x version is also supported. For cgroups v2 support, the recommended version is 5.10 or newer

	[WARNING Service-Kubelet]: kubelet service is not enabled, please run 'systemctl enable kubelet.service'

error: error execution phase wait-control-plane: failed while waiting for the control plane to start: kube-apiserver check failed at https://192.168.58.2:8443/livez: Get "https://control-plane.minikube.internal:8443/livez?timeout=10s": dial tcp 192.168.58.2:8443: connect: connection refused

To see the stack trace of this error execute with --v=5 or higher
```

while the other tests pass. 😕

iPraveenParihar commented:
let me rebase this PR and try again 🤞

iPraveenParihar committed:
ci: add claimProvision timeout

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar commented:
/test ci/centos/k8s-e2e-external-storage/1.34

iPraveenParihar commented:
@Madhu-1, the failing CI has passed. PTAL again

Madhu-1 commented:
@iPraveenParihar lets rerun all the test again to ensure there is no flaky tests to avoid causing more problems after merging this PR?

iPraveenParihar commented:
@Madhu-1, all CI passed 🎉 .
PTAL thanks!
